test/end-to-end/core.sh

```diff
@@ -204,9 +209,32 @@ echo "[INFO] Docker login successful"
 # log back into docker as e2e-user again
 os::cmd::expect_success "docker login -u e2e-user -p ${e2e_user_token} -e e2e-user@openshift.com ${DOCKER_REGISTRY}"
 
+# verify cross-repo mount functionality
+echo "[INFO] Docker cross-repo mount"
+os::cmd::expect_success "oc new-project crossmount"
+os::cmd::expect_success "oc create imagestream repo"
+
 echo "[INFO] Back to 'default' project with 'admin' user..."
 os::cmd::expect_success "oc project ${CLUSTER_ADMIN_CONTEXT}"
 os::cmd::expect_success_and_text 'oc whoami' 'system:admin'
+os::cmd::expect_success "oc tag --source docker centos/ruby-22-centos7:latest -n custom ruby-22-centos7:latest"
+os::cmd::expect_success 'oc policy add-role-to-user registry-viewer pusher -n custom'
+
+os::cmd::expect_success_and_text "curl -I -X HEAD -u 'pusher:${pusher_token}' '${DOCKER_REGISTRY}/v2/crossmount/repo/blobs/$rubyimageblob'" "404 Not Found"
+os::cmd::expect_success_and_text "curl -I -X HEAD -u 'pusher:${pusher_token}' '${DOCKER_REGISTRY}/v2/cache/ruby-22-centos7/blobs/$rubyimageblob'" "200 OK"
+# 202 means that cross-repo mount has failed (in this case because of blob doesn't exist in the source repository), client needs to reupload the blob
+os::cmd::expect_success_and_text "curl -I -X POST -u 'pusher:${pusher_token}' '${DOCKER_REGISTRY}/v2/crossmount/repo/blobs/uploads/?mount=$rubyimageblob&from=cache/hello-world'" "202 Accepted"
```

Comment:
I don't think they cover it, we need blob tests for authenticated users,
and users with similar / overlapping namespace permissions.  We also need
to explicitly call out the cached permissions behavior with a test.

On Wed, Jul 20, 2016 at 8:31 AM, Michal Minar <notifications@github.com>
wrote:

> In test/end-to-end/core.sh
> <https://github.com/openshift/origin/pull/9819#discussion_r71515054>:
>
> >  echo "[INFO] Back to 'default' project with 'admin' user..."
> >  os::cmd::expect_success "oc project ${CLUSTER_ADMIN_CONTEXT}"
> >  os::cmd::expect_success_and_text 'oc whoami' 'system:admin'
> > +os::cmd::expect_success "oc tag --source docker centos/ruby-22-centos7:latest -n custom ruby-22-centos7:latest"
> > +os::cmd::expect_success 'oc policy add-role-to-user registry-viewer pusher -n custom'
> > +
> > +os::cmd::expect_success_and_text "curl -I -X HEAD -u 'pusher:${pusher_token}' '${DOCKER_REGISTRY}/v2/crossmount/repo/blobs/$rubyimageblob'" "404 Not Found"
> > +os::cmd::expect_success_and_text "curl -I -X HEAD -u 'pusher:${pusher_token}' '${DOCKER_REGISTRY}/v2/cache/ruby-22-centos7/blobs/$rubyimageblob'" "200 OK"
> > +# 202 means that cross-repo mount has failed (in this case because of blob doesn't exist in the source repository), client needs to reupload the blob
> > +os::cmd::expect_success_and_text "curl -I -X POST -u 'pusher:${pusher_token}' '${DOCKER_REGISTRY}/v2/crossmount/repo/blobs/uploads/?mount=$rubyimageblob&from=cache/hello-world'" "202 Accepted"
>
> Tests probably need to be both here and in the unit tests.
>
> Auth unit tests
> <https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/auth_test.go>
> seem sufficient to me. Additional negative tests have been just added by
> @liggitt <https://github.com/liggitt>'s d179662
> #diff-4ea7c8f0aaeae849c3760842d6b42c60R194
> <https://github.com/openshift/origin/commit/d1796620d753a09503ba7dd346c52f536c0b4434#diff-4ea7c8f0aaeae849c3760842d6b42c60R194>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9819/files/1742dd1c7e2f56e7e52207e5c024ed830ea84fca..29c2108e28e979ea31cba31a000179b8c1073bd5#r71515054>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p2Dy62mVK6zrq3BOM54Vd-LWd_0_ks5qXhUIgaJpZM4JLnQb>
> .
>